boot/boot_test.go

```diff
@@ -1048,13 +1060,10 @@ func (s *bootenv20Suite) TestHappyMarkBootSuccessfulKernelUpgradeRebootBeforeEna
 	c.Assert(err, IsNil)
 	c.Assert(vars, DeepEquals, map[string]string{"kernel_status": boot.DefaultStatus})
 
-	// we should also still have TryKernel() == pc-kernel_2.snap, as well as
-	// trying == true, because the symlink is still there, the snapstate mgr
-	// is responsible for cleaning it up when it notices the boot with the try
-	// kernel failed - that is done a level above this test
```

Comment:
so this was false?  or is there missing code in SetNextBoot code? are we missing new tests in  managers_test.go?  see my comment there

Reply:
yes this was false, and see my other comment but I think we are missing tests there, but I'm not sure if we want to have all the permutations inside managers_test where it's very expensive, and instead maybe we should try something lighter weight inside boot test pkg